From 9695e025f8d400ff92d296d0fa1faa9e30110469 Mon Sep 17 00:00:00 2001
From: aidan2020sh <dan.ai@intel.com>
Date: Fri, 29 Apr 2022 14:08:41 +0800
Subject: [PATCH] UPSTREAM: [Upstream] RPLP upstream

1 this patch is mainly for RPLP upstream
2 updated latest ADLP device id support list
---
 README.md                                     |  4 +-
 .../linux/gen12/ddi/media_sysinfo_g12.cpp     | 55 ++++++++-----------
 2 files changed, 25 insertions(+), 34 deletions(-)

diff --git a/README.md b/README.md
index be6af2c1d7f2..e250622c316e 100755
--- a/README.md
+++ b/README.md
@@ -79,7 +79,7 @@ export LIBVA_DRIVER_NAME=iHD
 - KBLx (KBL: Kaby Lake, CFL: Coffe Lake, WHL: Whiskey Lake, CML: Comet Lake, AML: Amber Lake)
 - ICL (Ice Lake)
 - JSL (Jasper Lake) / EHL (Elkhart Lake)
-- TGLx (TGL: Tiger Lake, RKL: Rocket Lake, ADL-S/P/N: Alder Lake, RPL-S: Raptor Lake)
+- TGLx (TGL: Tiger Lake, RKL: Rocket Lake, ADL-S/P/N: Alder Lake, RPL-S/P: Raptor Lake)
 - DG1/SG1
 - Alchemist(DG2)/ATSM
 
@@ -203,7 +203,7 @@ or later, which officially supported C++11.
    - EHL/JSL: starting from kernel 5.8
    - TGL: starting from kernel 5.9
    - RKL/ADL-S/ADL-P: starting from kernel 5.17
-   - ADL-N/RPL-S: [drm-tip](https://cgit.freedesktop.org/drm-tip)
+   - ADL-N/RPL-S/RPL-P: [drm-tip](https://cgit.freedesktop.org/drm-tip)
    - DG1/SG1: [intel-gpu/kernel](https://github.com/intel-gpu/kernel)
 
 ##### (*) Other names and brands may be claimed as property of others.
diff --git a/media_driver/linux/gen12/ddi/media_sysinfo_g12.cpp b/media_driver/linux/gen12/ddi/media_sysinfo_g12.cpp
index f5da82388e1d..3b3d17d20fcd 100644
--- a/media_driver/linux/gen12/ddi/media_sysinfo_g12.cpp
+++ b/media_driver/linux/gen12/ddi/media_sysinfo_g12.cpp
@@ -436,56 +436,47 @@ static bool adlpGt2Device46A0 = DeviceInfoFactory<GfxDeviceInfo>::
 static bool adlpGt2Device46A1 = DeviceInfoFactory<GfxDeviceInfo>::
     RegisterDevice(0x46A1, &adlpGt2Info);
 
-static bool adlpGt2Device46A2 = DeviceInfoFactory<GfxDeviceInfo>::
-    RegisterDevice(0x46A2, &adlpGt2Info);
-
 static bool adlpGt2Device46A3 = DeviceInfoFactory<GfxDeviceInfo>::
     RegisterDevice(0x46A3, &adlpGt2Info);
-    
+
 static bool adlpGt2Device46A6 = DeviceInfoFactory<GfxDeviceInfo>::
     RegisterDevice(0x46A6, &adlpGt2Info);
-    
-static bool adlpGt2Device46A8 = DeviceInfoFactory<GfxDeviceInfo>::
-    RegisterDevice(0x46A8, &adlpGt2Info);
-    
-static bool adlpGt2Device4628 = DeviceInfoFactory<GfxDeviceInfo>::
-    RegisterDevice(0x4628, &adlpGt2Info);
-    
-    
+
+static bool adlpGt2Device4626 = DeviceInfoFactory<GfxDeviceInfo>::
+    RegisterDevice(0x4626, &adlpGt2Info);
+
 static bool adlpGt2Device46B0 = DeviceInfoFactory<GfxDeviceInfo>::
     RegisterDevice(0x46B0, &adlpGt2Info);
 
 static bool adlpGt2Device46B1 = DeviceInfoFactory<GfxDeviceInfo>::
     RegisterDevice(0x46B1, &adlpGt2Info);
 
-static bool adlpGt2Device46B2 = DeviceInfoFactory<GfxDeviceInfo>::
-    RegisterDevice(0x46B2, &adlpGt2Info);
-
 static bool adlpGt2Device46B3 = DeviceInfoFactory<GfxDeviceInfo>::
     RegisterDevice(0x46B3, &adlpGt2Info);
-    
 
-static bool adlpGt2Device46C0 = DeviceInfoFactory<GfxDeviceInfo>::
-    RegisterDevice(0x46C0, &adlpGt2Info);
+static bool adlpGt2Device46A8 = DeviceInfoFactory<GfxDeviceInfo>::
+    RegisterDevice(0x46A8, &adlpGt2Info);
 
-static bool adlpGt2Device46C1 = DeviceInfoFactory<GfxDeviceInfo>::
-    RegisterDevice(0x46C1, &adlpGt2Info);
+static bool adlpGt2Device4628 = DeviceInfoFactory<GfxDeviceInfo>::
+    RegisterDevice(0x4628, &adlpGt2Info);
 
-static bool adlpGt2Device46C2 = DeviceInfoFactory<GfxDeviceInfo>::
-    RegisterDevice(0x46C2, &adlpGt2Info);
+static bool rplpGt2DeviceA7A0 = DeviceInfoFactory<GfxDeviceInfo>::
+    RegisterDevice(0xA7A0, &adlpGt2Info);
 
-static bool adlpGt2Device46C3 = DeviceInfoFactory<GfxDeviceInfo>::
-    RegisterDevice(0x46C3, &adlpGt2Info);
-    
-static bool adlpGt2Device46AA = DeviceInfoFactory<GfxDeviceInfo>::
-    RegisterDevice(0x46AA, &adlpGt2Info);
-    
-static bool adlpGt2Device462A = DeviceInfoFactory<GfxDeviceInfo>::
-    RegisterDevice(0x462A, &adlpGt2Info);
+static bool rplpGt2DeviceA720 = DeviceInfoFactory<GfxDeviceInfo>::
+    RegisterDevice(0xA720, &adlpGt2Info);
 
-static bool adlpGt2Device4626 = DeviceInfoFactory<GfxDeviceInfo>::
-    RegisterDevice(0x4626, &adlpGt2Info);
+static bool rplpGt2DeviceA7A8 = DeviceInfoFactory<GfxDeviceInfo>::
+    RegisterDevice(0xA7A8, &adlpGt2Info);
+
+static bool rplpGt2DeviceA7A1 = DeviceInfoFactory<GfxDeviceInfo>::
+    RegisterDevice(0xA7A1, &adlpGt2Info);
+
+static bool rplpGt2DeviceA721 = DeviceInfoFactory<GfxDeviceInfo>::
+    RegisterDevice(0xA721, &adlpGt2Info);
 
+static bool rplpGt2DeviceA7A9 = DeviceInfoFactory<GfxDeviceInfo>::
+    RegisterDevice(0xA7A9, &adlpGt2Info);
 #endif
 
 #ifdef IGFX_GEN12_ADLN_SUPPORTED
-- 
2.35.3

